tests/js/client/shell/api/api-replication.js

```diff
@@ -0,0 +1,1373 @@
+              if (isCluster) {
+                // In cluster this is allowed, local it is not, cluster just ignores it.
+                isAllowed(res, collname, testParam);
+                // Assert that computedValues is still default value
+                validateProperties({}, collname, 2);
+              } else {
+                isDisallowed(ERROR_HTTP_BAD_PARAMETER.code, ERROR_BAD_PARAMETER.code, res, testParam);
+              }
+              break;
+            }
+            case "shardKeys": {
+              isDisallowed(ERROR_HTTP_BAD_PARAMETER.code, ERROR_BAD_PARAMETER.code, res, testParam);
+              break;
+            }
+            case "numberOfShards": {
+              if (isCluster) {
+                if (typeof ignoredValue === "number" && ignoredValue < 0) {
+                  // This is actually a VPack Error.
+                  // We should consider to make this a non-internal Error.
+                  isDisallowed(ERROR_HTTP_SERVER_ERROR.code, ERROR_INTERNAL.code, res, testParam);
+                } else {
+                  isAllowed(res, collname, testParam);
+                }
+              } else {
+                if (typeof ignoredValue === "number") {
+                  if (ignoredValue < 0) {
+                    // This is actually a VPack Error.
+                    // We should consider to make this a non-internal Error.
+                    isDisallowed(ERROR_HTTP_SERVER_ERROR.code, ERROR_INTERNAL.code, res, testParam);
+                  } else {
+                    isAllowed(res, collname, testParam);
+                  }
+                } else {
+                  isDisallowed(ERROR_HTTP_BAD_PARAMETER.code, ERROR_BAD_PARAMETER.code, res, testParam);
+                }
+              }
+              break;
+            }
+            case "schema": {
+              isDisallowed(ERROR_HTTP_BAD_PARAMETER.code, ERROR_VALIDATION_BAD_PARAMETER.code, res, testParam);
+              break;
+            }
+            case "replicationFactor": {
+              if (isCluster) {
+                if (typeof ignoredValue === "number") {
+                  // We take doubles for integers
+                  if (ignoredValue > 2) {
+                    // but they are too large to be allowed
+                    isDisallowed(ERROR_HTTP_SERVER_ERROR.code, ERROR_CLUSTER_INSUFFICIENT_DBSERVERS.code, res, testParam);
+                  } else {
+                    // or too low
+                    isDisallowed(ERROR_HTTP_BAD_PARAMETER.code, ERROR_BAD_PARAMETER.code, res, testParam);
+                  }
+                } else {
+                  // All others are ignored
+                  isAllowed(res, collname, testParam);
+                }
+              } else {
+                // Just ignore if we are not in cluster.
+                isAllowed(res, collname, testParam);
+              }
+              break;
+            }
+            case "writeConcern":
+            case "minReplicationFactor": {
```

Comment:
`replicationFactor` from above could also fit in here, the case is duplicated.

Reply:
Good spot!